### PR TITLE
allow hklmnpst symbol subscripts as unicode

### LINKED ISSUE #9047: allow latin subscripts for small hklmnpst
This was started but had errors in #8477. I left a [note there](https://github.com/sympy/sympy/pull/8477/files#discussion_r25183266) about how to potentially fix the error. With tests added for each letter, this should be easy to fix.

### PR DESCRIPTION
**Pull request by zanzibar7**
Fill in a few missing unicode subscripts.

fixes #9047

### CODE REVIEW

File: sympy/printing/pretty/pretty_symbology.py [hunk 1]
```diff
@@ -166,6 +166,9 @@ def xstr(*args):
 # latin subscripts
 for l in 'aeioruvx':
     sub[l] = LSUB(l)
+# subscripts from ftp://ftp.unicode.org/Public/6.3.0/ucd/UnicodeData.txt
+for n, l in enumerate('hklmnpst'):
```
Comment (smichr) [inline sympy/printing/pretty/pretty_symbology.py]: Perhaps these letters can just be added to the string in line 167: 'aeioruvx' -> 'aeioruvxhklmnpst' and the correct name will be supplied.

Comment (zanzibar7) [inline sympy/printing/pretty/pretty_symbology.py]: These new unicode subscript characters are available in unicode version 6.3.0.  Atleast on my install, the `unicodedata` python module is only at unicode 5.2.0.  Your suggestion is clearly the better idea in the long run, but doesn't work right now.  Is there a good way to handle this?

**Comment by smichr** [comment]
> but doesn't work right now. Is there a good way to handle this?

It only fails for 2.X; perhaps you can add TypeError to line 513: `except (KeyError, TypeError):`. But then you won't get a pretty subscript. 

How about just make this an XFAIL test: it passes in 3.X but fails (currently) in 2.X. And by adding the TypeError at least one will still get the non-pretty form instead of a Traceback.

**Comment by smichr** [comment]
Applying this:

```diff
diff --git a/sympy/printing/pretty/pretty_symbology.py b/sympy/printing/pretty/p
index 3f18a74..42cf37c 100644
--- a/sympy/printing/pretty/pretty_symbology.py
+++ b/sympy/printing/pretty/pretty_symbology.py
@@ -510,7 +510,7 @@ def pretty_list(l, mapping):
             if pretty is None:
                 try:  # match by separate characters
                     pretty = ''.join([mapping[c] for c in s])
-                except KeyError:
+                except (TypeError, KeyError):
                     return None
             result.append(pretty)
         return result
diff --git a/sympy/printing/pretty/tests/test_pretty.py b/sympy/printing/pretty/
index b698c00..9d74cb8 100644
--- a/sympy/printing/pretty/tests/test_pretty.py
+++ b/sympy/printing/pretty/tests/test_pretty.py
@@ -24,6 +24,9 @@
 from sympy.core.compatibility import u_decode as u
 from sympy.core.compatibility import range

+from sympy.utilities.pytest import XFAIL
+
+
 a, b, x, y, z, k = symbols('a,b,x,y,z,k')
 th = Symbol('theta')
 ph = Symbol('phi')
@@ -274,6 +277,9 @@ def test_upretty_subs_missingin_24():
     assert upretty( Symbol('F_v') ) == u('Fᵥ')
     assert upretty( Symbol('F_x') ) == u('Fₓ')

+
+@XFAIL  # fails in 2.X
+def test_sub_missing_issue_9047():
     assert upretty( Symbol('F_h') ) == u('F<U+2095>')
     assert upretty( Symbol('F_k') ) == u('F<U+2096>')
     assert upretty( Symbol('F_l') ) == u('F<U+2097>')
@@ -283,6 +289,7 @@ def test_upretty_subs_missingin_24():
     assert upretty( Symbol('F_s') ) == u('F<U+209B>')
     assert upretty( Symbol('F_t') ) == u('F<U+209C>')

+
 def test_upretty_modifiers():
     # Accents
     assert upretty( Symbol('Fmathring') ) == u('F̊')
```

Gives

```
sympy\printing\pretty\tests\test_pretty.py[73] C:\Users\leslie\sympy\sympy\print
ing\pretty\pretty_symbology.py:58: UserWarning: No 'LATIN SUBSCRIPT SMALL LETTER
 H' in unicodedata
No 'LATIN SUBSCRIPT SMALL LETTER K' in unicodedata
No 'LATIN SUBSCRIPT SMALL LETTER L' in unicodedata
No 'LATIN SUBSCRIPT SMALL LETTER M' in unicodedata
No 'LATIN SUBSCRIPT SMALL LETTER N' in unicodedata
No 'LATIN SUBSCRIPT SMALL LETTER P' in unicodedata
No 'LATIN SUBSCRIPT SMALL LETTER S' in unicodedata
No 'LATIN SUBSCRIPT SMALL LETTER T' in unicodedata

  warnings.warn(unicode_warnings)
f...............................
.........................................                                  [OK]

======= tests finished: 72 passed, 1 expected to fail, in 2.70 seconds ========
```

and this

```
$ /c/Python34/python bin/test t_pret
============================= test process starts =============================
executable:         c:\Python34\python.exe  (3.4.1-final-0) [CPython]
architecture:       32-bit
cache:              yes
ground types:       python
random seed:        8058100
hash randomization: on (PYTHONHASHSEED=1464333327)

sympy\printing\pretty\tests\test_pretty.py[73] X...............................
.........................................                                  [OK]

________________________________ xpassed tests ________________________________
sympy\printing\pretty\tests\test_pretty.py: test_sub_missing_issue_9047

== tests finished: 72 passed, 1 expected to fail but passed, in 3.17 seconds ==
```

**Pull request closed without merging**